safe_transaction_service/events/services/queue_service.py

```diff
@@ -76,36 +76,56 @@ def get_queue_service():
 class QueueService:
     def __init__(self):
         self._connection_pool: List[BrokerConnection] = []
+        self._total_connections: int = 0
         self.unsent_events: List = []
 
-    def get_connection(self) -> BrokerConnection:
+    def get_connection(self) -> Optional[BrokerConnection]:
         """
         :return: A `BrokerConnection` from the connection pool if there is one available, othwerwise
             returns a new BrokerConnection
         """
+        if (
+            settings.EVENTS_QUEUE_POOL_CONNECTIONS_LIMIT
+            and self._total_connections >= settings.EVENTS_QUEUE_POOL_CONNECTIONS_LIMIT
+        ):
+            logger.warning(
+                "No available pool connections, currently using: %i",
```

Comment:
```suggestion
                "Number of active connections reached the pool limit: %d",
```

`%d` is recommended as `%i` might be deprecated